src/models/trainerMatchingDao.js

```diff
@@ -0,0 +1,190 @@
+        p.price AS price,
+        p.content AS content,
+        p.created_at AS createdAt ,
+        u.gender AS gender,
+        t.specialized AS specialized,
+        u.img_url AS imgUrl
+        FROM products p 
+        JOIN trainers t ON t.id = p.trainer_id
+        JOIN users u ON u.id = t.user_id
+        WHERE p.status = 1
+        AND u.id = ?
+        AND p.id = ?;
+        `,
+    [userId, productsId]
+  );
+  return result;
+};
+
+const isTrainer = async (userId) => {
+  const [result] = await AppDataSource.query(
+    `SELECT CASE
+        WHEN (SELECT u.nickname FROM trainers t
+            JOIN users u ON t.user_id  = u.id
+            WHERE u.id = ?) IS NULL
+            THEN 'false'
+            ELSE 'true'
+        END AS isTrainer;
+    `,
+    [userId]
+  );
+  return result.isTrainer;
+};
+// 구독자 여부 확인
+const isSubscribed = async (userId) => {
+  const [result] = await AppDataSource.query(
+    `SELECT CASE
+        WHEN (SELECT u.nickname FROM sub_orders so
+            JOIN users u ON so.user_id  = u.id
+            WHERE u.id = ?) IS NULL
+            THEN 'false'
+            ELSE 'true'
+        END AS isSubscribed;
+      `,
+    [userId]
+  );
+  return result.isSubscribed;
+};
+const isPostedTrainer = async (productsId, trainerId) => {
+  const [result] = await AppDataSource.query(
+    `SELECT CASE
+        WHEN (SELECT p.id FROM p
+            WHERE p.id = ?
+            AND p.trainer_id = ?) IS NULL
+            THEN 'false'
+            ELSE 'true'
+        END AS isPostedTrainer;
+          `,
+    [productsId, trainerId]
+  );
+  return result.isPostedTrainer;
+};
+
+const findTrainerId = async (userId) => {
+  const [trainerId] = await AppDataSource.query(
+    `
+    SELECT t.id AS id
+    FROM trainer t
+    JOIN users u ON u.id = t.user_id
+    WHERE u.id = ?;`,
+    [userId]
+  );
+  return trainerId.id;
+};
+const findCategoryName = async (trainerId) => {
+  const [category] = await AppDataSource.query(
+    `
+    SELECT wc.category AS name
+    FROM workout_category wc
+    JOIN trainers t ON t.specialized = wc.id
+    WHERE t.id = ?;`,
+    [trainerId]
+  );
+  return category.name;
+};
+const postTrainerMatching = async (
+  userId,
+  trainerId,
+  imgUrl,
+  availableArea,
+  price,
+  availableTime,
+  term,
+  content
+) => {
+  await AppDataSource.query(`
+  UPDATE users 
+  SET img_url = ${imgUrl}
+  WHERE id = ${userId};
+  `);
+
+  await AppDataSource.query(
+    `
+  INSERT INTO products ( 
+    trainer_id,
+    available_area,
+    available_time,
+    category_name,
+    term,
+    price,
+    content)
+  VALUES (?, ?, ?, ?, ?, ?, ?);`,
+    [
+      trainerId,
+      availableArea,
+      availableTime,
+      categoryName,
+      term,
+      price,
+      content,
+    ]
+  );
+};
+const deleteTrainerMatching = async (productsId, stat) => {
+  await AppDataSource.query(`
+    UPDATE producsts
+        status  = ${stat}
+        WHERE id = ${productsId};
+    `);
+};
```

Comment:
```suggestion
const updateTrainerMatchingStatus = async (productId, status) => {
  await AppDataSource.query(`
    UPDATE producsts
        status  = ${status}
        WHERE id = ${productsId};
    `);
};
```
1. 꼭 status가 삭제로 변경되는 상황이 아닌 경우에도 쓰일 수 있도록 확장성 있게 작성하셨네요 👍 그렇기 때문에 함수 이름도 delete 경우에만 국한되지 않도록 update로 사용하셔도 좋겠습니다.
2. status...가 그렇게 긴 변수명은 아니기 때문에, 원래 단어 그대로 써주셔도 좋을 것 같습니다 :)